api/predict.py

```diff
@@ -54,9 +60,12 @@ def post(self):
         try:
             preds = self.model_wrapper.predict(input_json)
         except:  # noqa
-            abort(400, "Invalid input to Pinyin converter, please check!")
+            abort(400, "The input word(s) is/are not valid utf-8 encoded Chinese word(s)")
```

Comment:
Can we catch a more specific exception here? What about errors that may actually call for a `500` response or some other invalidity in input?

Reply:
The code for some reason doesn't return UnicodeDecodeError. If you are able to find a particular input for which this error is raised, I can surely add it.

Reply:
For now I've added it.

Reply:
Hmm, pypinyin seems to skip anything that is non-Chinese without raising errors. Perhaps it is misleading here to say that the error is caused by invalid utf-8 encoding.

Reply:
pypinyin doesn't raise any error, the real error is raised by the downstream code in the dimsim package when it tries to iterate through the pinyin strings and doesn't find it. I can theoretically change/fix that as I own the dimsim code too but that would be a version update on PyPi. 
For now if you know of a better way to check inputs for utf-8 encodings and/or Mandarin chinese sanity checking, I'd be glad to add that.

Reply:
For UTF-8, I would say something like `UnicodeEncodeError` or `UnicodeDecodeError` might be raised, depending on the context. For Chinese character detection, I would use some sort of [range detection](https://stackoverflow.com/a/37311125/1150462). But this is a CJK detection method (because Traditional/Simplified Chinese, Japanese, and Korean share a lot of characters) and it looks like UTF-8 puts CJK characters in the same range -- I doubt whether there actually exists a reliable Chinese character detector without enumerating them.

Reply:
For now I have added both TypeError and UnicodeDecodeError.